Jenkinsfile

```diff
@@ -0,0 +1,144 @@
+ *    code must carry prominent notices stating that the original code was
+ *    changed and the date of the change.
+ *
+ *  4. All publications or advertising materials mentioning features or use of
+ *     this software are asked, but not required, to acknowledge that it was
+ *     developed by Intel Corporation and credit the contributors.
+ *
+ * 5. Neither the name of Intel Corporation, nor the name of any Contributor
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+// To use a test branch (i.e. PR) until it lands to master
+// I.e. for testing library changes
+//@Library(value="pipeline-lib@your_branch") _
+
+pipeline {
+    agent none
+
+    stages {
+        stage('Cancel Previous Builds') {
+            when { changeRequest() }
+            steps {
+                cancelPreviousBuilds()
+            }
+        }
+        stage('Build') {
+            parallel {
+                stage('Build on SLES 12.3') {
+                    when { beforeAgent true
+                           environment name: 'SLES12_3_DOCKER', value: 'true' }
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.sles.12.3'
+                            label 'docker_runner'
+                            args '--privileged=true'
+                            additionalBuildArgs '--build-arg UID=$(id -u)'
+                        }
+                    }
+                    steps {
+                        sh '''rm -rf artifacts/sles12.3/
+                              mkdir -p artifacts/sles12.3/
+                              make source
+                              cd _topdir/SOURCES; sudo build \
+                                --repo http://cobbler/cobbler/ks_mirror/SLES-12.3-x86_64/suse \
+                                --repo http://cobbler/cobbler/repo_mirror/sdk-sles12.3-x86_64 \
+                                --repo http://cobbler/cobbler/repo_mirror/updates-sles12.3-x86_64 \
+                                --repo http://cobbler/cobbler/repo_mirror/sdkupdate-sles12.3-x86_64 \
+                                --dist sles12.3'''
```

Comment:
This should be part of the previously mentioned recipe in the `Makefile_packaging.mk`.  Note that `build` can take an SRPM as what it builds and we already have a recipe to build the SRPM so that should fit in quite nicely so we don't need anything new to make sure the main source file is available for `build`.

I know I had not done this yet (integrate `build` into the Makefile) but this is all still a WIP for me.  I have not gotten far enough to do that integration due to other priorities.

Reply:
Until the nexus repo server is setup to to create a "repo group" the repository URLs are going to vary with what RPM needs to be built.

For some reason build on a real SLES system seems to be able to use the repositories configured for it with out needing them in the command line, but build in a docker container needs all the repositories specified.

So optimizing this should wait until after the Nexus system is integrated with Jenkins.

Reply:
They don't need to vary.  They can just be the total sum of all repositories needed to build everything.  Having a repository present that is not used should not cause any issues.

I am currently working on the stack of updates for the updated SPDK.  I will integrate build within the `Makefile_packaging.mk` as I make progress on that.